zaza/openstack/charm_tests/mysql/tests.py

```diff
@@ -0,0 +1,382 @@
+        return value
+
+    def is_pxc_bootstrapped(self):
+        """Determine if the cluster is bootstrapped.
+
+        Query the wsrep_ready status in the DB.
+
+        :returns: boolean
+        """
+        value = self.get_wsrep_value("wsrep_ready")
+        return value.lower() in ["on", "ready"]
+
+    def get_cluster_size(self):
+        """Determine the cluster size.
+
+        Query the wsrep_cluster size in the DB.
+
+        :returns: str Numeric cluster size
+        """
+        return self.get_wsrep_value("wsrep_cluster_size")
+
+    def get_crm_master(self):
+        """Determine CRM master for the VIP.
+
+        Query CRM to determine which node hosts the VIP.
+
+        :returns: str Unit name
+        """
+        for unit in zaza.model.get_units(self.application):
+            # is the vip running here?
+            cmd = 'sudo ip a | grep "inet {}/"'.format(self.vip)
+            result = zaza.model.run_on_unit(unit.entity_id, cmd)
+            code = result.get("Code")
+            output = result.get("Stdout").strip()
+            logging.info("Checking {}".format(unit.entity_id))
+            logging.debug(output)
+            if code == "0":
+                logging.info("vip ({}) running in {}".format(
+                    self.vip,
+                    unit.entity_id)
+                )
+                return unit.entity_id
+
+    def is_port_open(self, port="3306", address=None):
+        """Determine if MySQL is accessible.
+
+        Connect to the MySQL port on the VIP.
+
+        :param port: Port number
+        :type port: str
+        :param address: IP address
+        :type port: str
+        :returns: boolean
+        """
+        try:
+            telnetlib.Telnet(address, port)
+            return True
+        except socket.error as e:
+            if e.errno == 113:
+                self.log.error("could not connect to %s:%s" % (address, port))
+            if e.errno == 111:
+                self.log.error("connection refused connecting"
+                               " to %s:%s" % (address,
+                                              port))
+            return False
+
+    def update_leaders_and_non_leaders(self):
+        """Get leader node and non-leader nodes of percona.
+
+        Update and set on the object the leader node and list of non-leader
+        nodes.
+
+        :side effect: sets self.leader and self.non_leaders
+        :returns: None
+        """
+        status = zaza.model.get_status().applications[self.application]
+        # Reset
+        self.leader = None
+        self.non_leaders = []
+        for unit in status["units"]:
+            if status["units"][unit].get("leader"):
+                self.leader = unit
+            else:
+                self.non_leaders.append(unit)
+
+
+class PerconaClusterCharmTests(PerconaClusterTest):
+    """Base for percona-cluster charm tests.
+
+    Note: these have tests have been ported from amulet tests
```

Comment:
If you want the Note to call out, you can use:

.. note:: these have tests ...

A formatter will then pull this out like an rst document.